docs/rollout.md

```diff
@@ -0,0 +1,175 @@
+
+Various limits that can be configured in Fleet:
+
+| Field | Description | Default |
+| ----- | ----- | ----- |
+| maxUnavailable | Maximum number or percentage of clusters that can be non-ready before halting rollout. | 100% |
+| maxUnavailablePartitions | Number or percentage of partitions that can be non-ready at once. | 0 |
+| autoPartitionSize | Number or percentage of clusters per auto-created partition. | 25% |
+| partitions | Define manual partitions by cluster labels or group. If set, autoPartitionSize is ignored. | – |
+
+Fleet supports both automatic and manual rollout partitioning. For more information, refer to the [rolloutStrategy in  fleet.yaml.](ref-fleet-yaml.md)
+
+**Automatic Partitioning:** Fleet automatically creates partitions using autoPartitionSize.
+
+* If fewer than 200 clusters, Fleet uses a single partition.  
+* If 200+ clusters, partitions are created based on autoPartitionSize.
+
+**Manual Partitioning:** You define specific partitions using the partitions field. This gives you fine-grained control over cluster groupings and rollout order.
+
+:::note
+If you specify partitions manually, the autoPartitionSize is ignored.
+:::
+
+Partitions can be specified by name, `cluster selector`, `cluster group`, and `clusterGroupSelector`.
+
+For example, consider creating a partition named canary for production clusters in a specific group:
+
+```yaml
+rolloutStrategy:
+  partitions:
+    - name: canary
+      maxUnavailable: 10%
+      clusterSelector:
+        matchLabels:
+          env: prod
+      clusterGroup: frontend
+```
+
+Fleet then:
+
+1. Selects clusters based on `clusterSelector`, `clusterGroup`, or `clusterGroupSelector`.  
+2. Starts rollout to the first partition.  
+3. Waits until the partition is **Ready** (or within maxUnavailable threshold).  
+4. Proceeds to the next partition.
+
+### Single Partition Rollout
+
+If you do not define partitions, Fleet creates a single partition with all clusters. Use the `maxUnavailable` setting to limit how many clusters can be non-ready at once. If this limit is exceeded, Fleet pauses the rollout until enough clusters become ready.
+
+Fleet automatically creates a single partition only for clusters \<200, and then it is 25% of all clusters, because that is the default value for autoPartitionSize.
+
+### Multiple Partitions Rollout
+When you define multiple partitions, Fleet uses maxUnavailablePartitions to limit how many partitions can be non-ready at once. If the number of non-ready partitions exceeds maxUnavailablePartitions, Fleet pauses the rollout.
+
+
+## Preventing image pull storms
+
+During rollout, each downstream cluster pulls container images. If Fleet triggers updates on hundreds of clusters at once, all may start pulling images simultaneously—causing registry overload or DDoS-like behavior.
+
+To mitigate this Fleet uses a reconciliation loop to progressively rollout workloads. It is staged using:
```

Comment:
We should probably not mention "reconciliation loop". Something internal people shouldn't need to be aware of.

What about saying that we can slow down how many things are happening a the same time by using the options mentioned below (and perhaps be a bit more specific with a use case)?